inc/azure_c_shared_utility/tlsio.h

```diff
@@ -16,6 +16,7 @@ typedef struct TLSIO_CONFIG_TAG
     int port;
     const IO_INTERFACE_DESCRIPTION* underlying_io_interface;
     void* underlying_io_parameters;
+    const char* protocol;
```

Comment:
This could be expressed as a bool

```suggestion
    const bool is_http_protocol;
```